torch/csrc/jit/script/init.cpp

```diff
@@ -251,12 +260,37 @@ struct ModuleValue : public SugaredValue {
     // This can also be a call to a non-script module, or a plain
     // python method. If so return this as a python value.
     py::object py_module = py::cast(module);
+
     if (py::object attr = py::getattr(py_module, field.c_str(), py::none())) {
       if (py::isinstance<py::function>(attr) ||
           py::isinstance(attr, py::module::import("torch.nn").attr("Module")) ||
           py_module.attr("_constants_set").contains(field.c_str())) {
         return toSugaredValue(attr, m, loc, true);
       } else {
+        if (py::isinstance(attr, py::module::import("torch").attr("Tensor"))) {
```

Comment:
Users are going to expect that like other uses of @property, that they can put anything in these property methods that is valid torchscript. Really a property is just just syntax sugar that says take `a.foo` to mean `a.foo()`.

 Restricting this to a special cases here adds an entire separate pathway that we will have to service with feature enhancements for properties.

 I think it would be much more sane if @property (or @script_property) was just syntax sugar for invoking the script function underneath.